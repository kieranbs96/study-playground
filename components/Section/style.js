import React from 'react';
import styled from 'styled-components';

export const Block = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 5.6rem;
  padding-bottom: 4rem;
  border-bottom: 1px dotted #333;

  &.text-section {

    div {
      display: flex;
      flex-direction: column;
      flex: 1;
    }

    img {
      flex: 1;
      margin-left: 1rem;
    }
  }

  &.gallery-section {
    flex-direction: column;

    img {
      order: 1;

    }

    div {
      order: 2;
    }
  }
`;