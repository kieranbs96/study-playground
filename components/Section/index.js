import React, { Component } from 'react';
import { Block } from './style';

class Section extends Component {
  render() {
    const {
      sectionData: { title, type, description, image },
    } = this.props;

    return (
      <Block type={type} className={type}>
        <div>
          <h3>{title}</h3>
          <p>{description}</p>
        </div>
        <img src={image} />
      </Block>
    );
  }
}

export default Section;
