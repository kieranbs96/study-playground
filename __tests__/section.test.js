import React from 'react';
import Section from '../components/Section';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const sectionData = {
    title: 'Section One',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam lacinia dapibus purus, quis mollis dui eleifend id. Cras nec libero id elit congue pharetra. Phasellus laoreet metus eu leo porttitor commodo. Integer vestibulum mi sit amet nisl vulputate, sed porttitor lectus rutrum. Cras elit mi, lobortis ac elementum quis, pretium gravida magna. Vestibulum pulvinar arcu nec massa ultrices blandit. Phasellus laoreet vulputate diam, at imperdiet risus consequat id. Etiam ligula tortor, auctor sed suscipit at, auctor sit amet libero. Pellentesque ac pretium eros.',
    image: 'https://i.picsum.photos/id/776/960/320.jpg',
    type: 'gallery-section'
};

describe('<Section/>', () => {
  it('renders the title and description properly', () => {
    const wrapper = shallow(<Section sectionData={sectionData} />);
    const Title = wrapper.find('h3');
    const Description = wrapper.find('p');

    expect(Title.text()).toBe(sectionData.title);
    expect(Description.text()).toBe(sectionData.description);
  });

  it('renders the image properly', () => {
    const wrapper = shallow(<Section sectionData={sectionData} />);
    const Image = wrapper.find('img');
    expect(Image.props().src).toBe(sectionData.image);
  })

  it('matches the snapshot', () => {
    const wrapper = shallow(<Section sectionData={sectionData} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    // const price = 'Â£50.35';
    // expect(price).toMatchSnapshot();
  })
});
