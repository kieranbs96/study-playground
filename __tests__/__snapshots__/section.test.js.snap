// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section/> matches the snapshot 1`] = `
<styled.div
  className="gallery-section"
  type="gallery-section"
>
  <div>
    <h3>
      Section One
    </h3>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam lacinia dapibus purus, quis mollis dui eleifend id. Cras nec libero id elit congue pharetra. Phasellus laoreet metus eu leo porttitor commodo. Integer vestibulum mi sit amet nisl vulputate, sed porttitor lectus rutrum. Cras elit mi, lobortis ac elementum quis, pretium gravida magna. Vestibulum pulvinar arcu nec massa ultrices blandit. Phasellus laoreet vulputate diam, at imperdiet risus consequat id. Etiam ligula tortor, auctor sed suscipit at, auctor sit amet libero. Pellentesque ac pretium eros.
    </p>
  </div>
  <img
    src="https://i.picsum.photos/id/776/960/320.jpg"
  />
</styled.div>
`;
